llvm/lib/Transforms/IPO/AttributorAttributes.cpp

```diff
@@ -12592,19 +12592,38 @@ struct AAAddressSpaceImpl : public AAAddressSpace {
 
   ChangeStatus updateImpl(Attributor &A) override {
     uint32_t OldAddressSpace = AssumedAddressSpace;
+    unsigned FlatAS = A.getInfoCache().getFlatAddressSpace().value();
 
     auto CheckAddressSpace = [&](Value &Obj) {
-      if (isa<UndefValue>(&Obj))
+      // Ignore undef and poison.
+      if (isa<UndefValue>(&Obj) || isa<PoisonValue>(&Obj))
         return true;
```

Comment:
`isa<UndefValue>` already covers undef and poison. However, I think in this context it's unsafe to handle undef. You shouldn't touch this in this PR